docs/api-reference/specifications/loader-object.md

```diff
@@ -0,0 +1,58 @@
+# Loader Object Specification
+
+To be compatible with loaders.gl functions, loader objects need to conform to the following specification:
+
+## v1.0 Loader Object
+
+### Common Fields
+
+| Field           | Type        | Default    | Description |
+| ---             | ---         | ---        | ---         |
+| `name`          | `String`    | Required   | Short name of the loader ('OBJ', 'PLY' etc) |
+| `extension`     | `String`    | Required   | Three letter (typically) extension used by files of this format |
+| `category`      | `String`    | Optional   | Indicates the type/shape of data |
+
+
+### Test Function
+
+| Field           | Type        | Default    | Description |
+| ---             | ---         | ---        | ---         |
+| `testText`      | `Function`  | `null`     | Guesses if a file is of this format by examining the first characters in the file |
+
+
+### Parser Function
+
+When creating a new loader object, at least one of the parser functions needs to be defined:
+
+| Parser function field | Type        | Default    | Description |
+| ---                   | ---         | ---        | ---         |
+| `parseTextSync`       | `Function`  | `null`     | Parses a text file synchronously (`String`) |
+| `parseSync`           | `Function`  | `null`     | Parses a binary file synchronously (`ArrayBuffer`) |
+| `parse`               | `Function`  | `null`     | Parses a binary file asynchronously (`ArrayBuffer`) |
```

Comment:
The word "file" is ambiguous here. Maybe:

`parse*`: Parses a binary/text *object* ...
vs.
`load`: Reads binary from a file and parse ...